### PR TITLE
Query Tasks Based on Timing

### PR DESCRIPTION
**Pull request by moshthepitt**
- [x] Add TaskOccurrences
- [x] Add a way to generate occurrences
- [x] Add occurrence filters to TaskViewSet

Fix: #10

### CODE REVIEW

File: tests/serializers/test_occurrences.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            'date': '2018-05-24',
+            'start_time': '07:00',
+            'end_time': '19:00'
+        }
+
+        serializer_instance = TaskOccurrenceSerializer(data=data)
+        self.assertTrue(serializer_instance.is_valid())
+        occurrence = serializer_instance.save()
+
+        # response.data will return the time fields with seconds
```
[the remark this answers is not in the document]
Comment (moshthepitt) [reply tests/serializers/test_occurrences.py]: you are right

File: tasking/utils.py [hunk 2]
```diff
@@ -4,29 +4,140 @@
...
-    Validates an rrule string; returns True or False
+    Generates TaskOccurrence objects using the Task timing_rule field
+
+    It works this way:
+        - gets the start_time from the timing_rule
+        - only generates a maximum og MAX_OCCURRENCES
+        - the end time is always 23:59:59
+            * the very last task occurrence will have the same end_time as
+              the end_time from the timing_rule
+        - occurrences with the same setart_time and end_time will not be
```
Comment (DavisRayM) [inline tasking/utils.py]: Did you mean " same start time" ?

Comment (moshthepitt) [reply tasking/utils.py]: Yes I did.  Now fixed

File: tasking/utils.py [hunk 3]
```diff
@@ -4,29 +4,140 @@
...
-    Validates an rrule string; returns True or False
+    Generates TaskOccurrence objects using the Task timing_rule field
+
+    It works this way:
+        - gets the start_time from the timing_rule
+        - only generates a maximum of MAX_OCCURRENCES
+        - the end time is always 23:59:59
+            * the very last task occurrence will have the same end_time as
+              the end_time from the timing_rule
+        - occurrences with the same tart_time and end_time will not be
```
Comment (DavisRayM) [inline tasking/utils.py]: Typo here i think, should probably be start_time

**Pull request merged**